src/Apropos/Gen.hs

```diff
@@ -135,40 +202,45 @@ retryLimit lim g done = go lim
     then pure ()
     else failWithFootnote $ "expected: " <> show l <> " === " <> show r
 
-genProp :: Gen a -> Property
-genProp g = H.property $ void $ runExceptT $ gen g
-
 data GenException = GenException String | Retry deriving stock (Show)
 
-type GenContext = PropertyT IO
+type GenContext = H.Gen
```

Comment:
Now our `GenContext` is the `Hedgehog` `Gen` monad we can gain control over the shrinker from within our `Gen` free monad. This should allow us to solve the issues with trivial generators misbehaving due to sizing operations messing with the expected distributions.